### PR TITLE
Log in to VMI serial console

### PR DESCRIPTION
**Pull request by orelmisan**
Try to log in to the VMI serial console, in order to perform the necessary commands required to perform the checkup.

The `console` code is based on code from KubeVirt e2e tests suite [1]

[1] https://github.com/kubevirt/kubevirt/tree/ef62d0f87a95580c2c449fa4bbd1c5a13d15d81a/tests/console

Manually tested against an OpenShift Virtualization 4.12 cluster.

~~Depends on PR #40.~~

### CODE REVIEW

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/4849008e52125830bf41ed8a078b426b2ebb102c..d9de8a3cb08cc75eb4879f6cae8da2347abd902a): Taken the namespace out of the `Execute()` function, so it could be shared between the VMI and the Pod.

**Review by RamLavi — CHANGES_REQUESTED** [review]

File: pkg/internal/checkup/console/login.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	"regexp"
+	"time"
+
+	expect "github.com/google/goexpect"
+	"google.golang.org/grpc/codes"
+
+	kvcorev1 "kubevirt.io/api/core/v1"
+)
+
+func LoginToCentOS(serialConsoleClient vmiSerialConsoleClient, vmi *kvcorev1.VirtualMachineInstance, username, password string) error {
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply pkg/internal/checkup/console/login.go]: Yes.

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/d9de8a3cb08cc75eb4879f6cae8da2347abd902a..05144b2afc31bdc0efbd07063f9d20c1eb102c9f): Rebase

File: pkg/internal/checkup/executor/executor.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	corev1 "k8s.io/api/core/v1"
+
+	kvcorev1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"github.com/kiagnose/kubevirt-dpdk-checkup/pkg/internal/status"
+)
+
+type vmiSerialConsoleClient interface {
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply pkg/internal/checkup/executor/executor.go:33]: I suggest to make the suggested change in the PR that will introduce the pod to the executor.
Right now, this is the minimal required functionality.

**Review by RamLavi — APPROVED** [review]
/lgtm

**Pull request merged**